nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -249,8 +249,8 @@ NIXOS_INSTALL_GRUB=1 chroot $mountPoint \
 chroot $mountPoint /nix/var/nix/profiles/system/activate
 
 
-# Ask the user to set a root password.
-if [ -t 0 ] ; then
+# Ask the user to set a root password if it's not set already
+if grep -q '^root:!' $mountPoint/etc/shadow && [ -t 0 ] ; then
```

Comment:
IMO, this should be done anyways so nixos-install can be run as a nixos-rebuld from recovery without having to type root password every time. (or if it didn't finish installing)